packages/quantic/force-app/main/default/lwc/quanticResultPrintableUri/__tests__/quanticResultPrintableUri.test.js

```diff
@@ -0,0 +1,327 @@
+  jest.spyOn(mockHeadlessLoader, 'getHeadlessEnginePromise').mockReturnValue(
+    new Promise((resolve) => {
+      resolve();
+    })
+  );
+
+  jest.spyOn(mockHeadlessLoader, 'getBueno').mockReturnValue(
+    new Promise(() => {
+      // @ts-ignore
+      global.Bueno = {
+        isString: jest
+          .fn()
+          .mockImplementation(
+            (value) =>
+              Object.prototype.toString.call(value) === '[object String]'
+          ),
+      };
+    })
+  );
+}
+
+const selectors = {
+  link: 'c-quantic-result-link',
+  pathItem: '[data-test="path__item"]',
+  pathExpandButton: '[data-test="path__expand-button"]',
+  errorMessage: '[data-test="error"]',
+};
+const exampleEngineId = 'example engine id';
+
+function createTestComponent(options) {
+  const element = createElement('c-quantic-result-printable-uri', {
+    is: QuanticResultPrintableUri,
+  });
+
+  for (const [key, value] of Object.entries(options)) {
+    element[key] = value;
+  }
+
+  document.body.appendChild(element);
+  return element;
+}
+
+// Helper function to wait until the microtask queue is empty.
+function flushPromises() {
+  return new Promise((resolve) => setTimeout(resolve, 0));
+}
+
+function generateExampleUri(id) {
+  return `https://uri${id}.com/`;
+}
+
+function generateResultParents(numberOfParents) {
+  let parents = '';
+  for (let i = 0; i < numberOfParents; i++) {
+    parents += `<parent name="parent${i + 1}" uri="${generateExampleUri(i + 1)}" />`;
+  }
+  return `<parents>${parents}</parents>`;
+}
+
+describe('c-quantic-result-printable-uri', () => {
+  function cleanup() {
+    // The jsdom instance is shared across test cases in a single file so reset the DOM
+    while (document.body.firstChild) {
+      document.body.removeChild(document.body.firstChild);
+    }
+    jest.clearAllMocks();
+  }
+
+  beforeEach(() => {
+    console.error = jest.fn();
+    mockHeadlessAndBueno();
+  });
+
+  afterEach(() => {
+    cleanup();
+  });
+
+  describe('when displaying the result path', () => {
+    describe('when the number of parents is lower than the maxNumberOfParts property', () => {
+      const numberOfParents = 2;
+      const exampleResultWithParents = {
+        raw: {
+          parents: generateResultParents(numberOfParents),
+        },
+      };
+      it('should properly display all the result path', async () => {
+        const element = createTestComponent({
+          result: exampleResultWithParents,
+          engineId: exampleEngineId,
+          maxNumberOfParts: 3,
+        });
+        await flushPromises();
+
+        const pathItems = element.shadowRoot.querySelectorAll(
+          selectors.pathItem
+        );
+
+        expect(pathItems).not.toBeNull();
+        expect(pathItems.length).toBe(numberOfParents);
+        pathItems.forEach((item, index) => {
+          expect(item.textContent).toBe(`parent${index + 1}`);
+          expect(item.href).toBe(`${generateExampleUri(index + 1)}`);
+        });
```

Comment:
Would it be possible to regroup this in one single function then? In order to simplify the test file a lot?

```
expect(pathItems).not.toBeNull();
validatePathItems(pathItems, numberOfParents);

function validatePathItems(pathItems, expectedNumberOfParents) {
    expect(pathItems.length).toBe(expectedNumberOfParents);
    pathItems.forEach((item, index) => {
        expect(item.textContent).toBe(`parent${index + 1}`);
        expect(item.href).toBe(`${generateExampleUri(index + 1)}`);
    });
  return true;
}
```

I haven't looked at the entire test file, but it feels like it could be improved like this. And you could limit a lot the places where you have to call `generateExampleUri` ?

Reply:
It's more like a question of preference on how we wanna read the tests, I added this function tho